pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/analytics/PlaybackSessionManager.kt

```diff
@@ -169,36 +190,37 @@ class PlaybackSessionManager {
 
     private inner class SessionManagerAnalyticsListener : PillarboxAnalyticsListener {
         override fun onPositionDiscontinuity(
-            eventTime: AnalyticsListener.EventTime,
+            eventTime: EventTime,
             oldPosition: Player.PositionInfo,
             newPosition: Player.PositionInfo,
             @DiscontinuityReason reason: Int
         ) {
             val oldItemIndex = oldPosition.mediaItemIndex
             val newItemIndex = newPosition.mediaItemIndex
 
+            if (eventTime.timeline.isEmpty) return
             DebugLogger.debug(TAG, "onPositionDiscontinuity reason = ${StringUtil.discontinuityReasonString(reason)}")
-
             if (oldItemIndex != newItemIndex && !eventTime.timeline.isEmpty) {
-                currentSession = getOrCreateSession(eventTime)
+                val newSession = getOrCreateSession(eventTime)!! // Return null only if timeline is empty
+                setCurrentSession(SessionInfo(newSession, newPosition.positionMs), oldPosition.positionMs)
             }
         }
 
         override fun onMediaItemTransition(
-            eventTime: AnalyticsListener.EventTime,
+            eventTime: EventTime,
             mediaItem: MediaItem?,
             @MediaItemTransitionReason reason: Int,
         ) {
             DebugLogger.debug(
                 TAG,
                 "onMediaItemTransition reason = ${StringUtil.mediaItemTransitionReasonString(reason)} ${mediaItem?.mediaMetadata?.title}",
             )
-
-            currentSession = mediaItem?.let { getOrCreateSession(eventTime) }
+            // Already handled in position discontinuity?
+            // currentSession = mediaItem?.let { getOrCreateSession(eventTime) }
```

Comment:
So maybe we can remove the `onMediaItemTransition` method?

Reply:
Totally ;)